packages/datadog-plugin-mongodb-core/src/index.js

```diff
@@ -55,8 +55,8 @@ class MongodbCorePlugin extends Plugin {
 
 function getQuery (cmd) {
   if (!cmd || typeof cmd !== 'object' || Array.isArray(cmd)) return
-  if (cmd.query) return JSON.stringify(sanitize(cmd.query))
-  if (cmd.filter) return JSON.stringify(sanitize(cmd.filter))
+  if (cmd.query) return truncate(JSON.stringify(limitDepth(cmd.query)))
```

Comment:
Truncating would prevent parsing and effectively result in no query at all. Let's limit the depth more instead. Not sure what the magic number is, but I'd be very surprised to see queries above for example 10 levels of depth.

Reply:
Actually, maybe the rule can be different for the resource name and the tag. Truncating the resource name could make sense but keeping the whole query for the tag. I'd still reduce the depth in both cases.